### PR TITLE
Improve current logging for SATOSA

### PR DESCRIPTION
**Pull request by sebulibah**
### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [ ] Have you added an explanation of what problem you are trying to solve with this PR?
* [ ] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?
* [ ] Does your submission pass tests?
* [ ] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

**Comment by c00kiemon5ter** [comment]
I'm taking a random change:

```diff
-        satosa_logging(logger, logging.DEBUG, "Routing to backend: %s " % context.target_backend, context.state)
+        logger.debug("Routing to backend: %s " % context.target_backend)
```

---

I note two things here:

1. To concatenate strings we should prefer `.format()`

```diff
-        satosa_logging(logger, logging.DEBUG, "Routing to backend: %s " % context.target_backend, context.state)
+        msg = "Routing to backend: {backend}".format(backend=context.target_backend)
+        logger.debug(msg)
```

2. The chosen change does not include `context.state` that was previously part of the content that was logged. Specifically, if we look at the `satosa_logging()` function, we see that _state_ is only used to extract the _session id_. So, let's try to keep that:

```diff
-        satosa_logging(logger, logging.DEBUG, "Routing to backend: %s " % context.target_backend, context.state)
+        msg = "Routing to backend: {backend}".format(backend=context.target_backend)
+        logline = "[{id}] {message}".format(id=state.get("SESSION_ID"), message=msg)
+        logger.debug(logline)
```

File: src/satosa/routing.py [hunk 1]
```diff
@@ -80,7 +79,9 @@ def backend_routing(self, context):
         :param context: The request context
         :return: backend
         """
-        satosa_logging(logger, logging.DEBUG, "Routing to backend: %s " % context.target_backend, context.state)
+        msg =  "Routing to backend: {backend}".format(backend=context.target_backend)
+        logline = "[{id}] {message}".format(id=context.state, message=msg)
```
Comment (c00kiemon5ter) [inline src/satosa/routing.py]: This should be `context.state.get("SESSION_ID")`

File: src/satosa/routing.py [hunk 2]
```diff
@@ -97,7 +98,9 @@ def frontend_routing(self, context):
         """
 
         target_frontend = context.state[STATE_KEY]
-        satosa_logging(logger, logging.DEBUG, "Routing to frontend: %s " % target_frontend, context.state)
+        msg = "Routing to frontend: {frontend} ".format (frontend=target_frontend)
```
Comment (c00kiemon5ter) [inline src/satosa/routing.py]: No need for this extra whitespace at the end, here.

File: src/satosa/routing.py [hunk 3]
```diff
@@ -136,17 +139,20 @@ def endpoint_routing(self, context):
         :return: registered endpoint and bound parameters
         """
         if context.path is None:
-            satosa_logging(logger, logging.DEBUG, "Context did not contain a path!", context.state)
+            logger.debug("Context did not contain a path!")
```
Comment (c00kiemon5ter) [inline src/satosa/routing.py]: There is some extra whitespace at the end of the line here.

Some calls are still missing the session id.

**Comment by c00kiemon5ter** [comment]
Thanks @sebulibah 

This is the first merge towards improving the state of logging in the project. For those interested we will start by removing the `satosa_logging` function, replacing it with direct calls to the logger. Then, we will switch to using, probably, the `structlog` library and starting taking advantage of the features of structured logging.

**Comment by peppelinux** [comment]
Good news, I prefer direct call to logging instances too

File: src/satosa/routing.py [hunk 4]
```diff
@@ -136,17 +141,24 @@ def endpoint_routing(self, context):
...
+        logger.debug(logline)
         path_split = context.path.split("/")
         backend = path_split[0]
 
         if backend in self.backends:
             context.target_backend = backend
         else:
-            satosa_logging(logger, logging.DEBUG, "Unknown backend %s" % backend, context.state)
+            msg = "Unknown backend {}".format(backend)
+            logline = "[{id} {message}".format(id=context.state.get("SESSION_ID"), message=msg)
```
Comment (c00kiemon5ter) [inline src/satosa/routing.py]: there was a typo here; the `id` should have been followed by `]`

```diff
- "[{id} {message}"
+ "[{id}] {message}"
```

**Comment by c00kiemon5ter** [comment]
@sebulibah see 053df986ae1e45a938d9de97495474c2aef30530 and 4065cfd7ac2c1802977408f5f8b57d83202bcd9e . 

I have extracted the log format and the way we should get the session id, so it should be easier to follow that pattern and avoid such typos in the future ;)

**Comment by sebulibah** [comment]
> Thanks @sebulibah
> 
> This is the first merge towards improving the state of logging in the project. For those interested we will start by removing the `satosa_logging` function, replacing it with direct calls to the logger. Then, we will switch to using, probably, the `structlog` library and start taking advantage of the features of structured logging.

You're welcome

**Pull request merged**